persistence/playlist_repository_test.go

```diff
@@ -143,5 +145,33 @@ var _ = Describe("PlaylistRepository", func() {
 				Expect(repo.Put(&newPls)).To(MatchError(ContainSubstring("invalid criteria expression")))
 			})
 		})
+
+		Context("child smart playlists", func() {
+			It("Should refresh tracks for smart playlist referenced in parent smart playlist criteria", func() {
+				conf.Server.SmartPlaylistRefreshTimeout = -1 * time.Second
+
+				nestedPls := model.Playlist{Name: "Nested", OwnerID: "userid", Rules: rules}
+				Expect(repo.Put(&nestedPls)).To(Succeed())
+
+				parentPls := model.Playlist{Name: "Parent", OwnerID: "userid", Rules: &criteria.Criteria{
+					Expression: criteria.All{
+						criteria.InPlaylist{"id": nestedPls.ID},
+					},
+				}}
+				Expect(repo.Put(&parentPls)).To(Succeed())
+
+				nestedPlsRead, err := repo.Get(nestedPls.ID)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = repo.GetWithTracks(parentPls.ID, true)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Check that the nested playlist was refreshed by parent get by verifying evaluatedAt is updated since first nestedPls get
+				nestedPlsAfterParentGet, err := repo.Get(nestedPls.ID)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(*nestedPlsAfterParentGet.EvaluatedAt).To(BeTemporally(">", *nestedPlsRead.EvaluatedAt))
```

Comment:
I think  it would be better to update the nested playlist and check the updated value is referenced correctly in the parent playlist, which is a more specific test about the expected behavior of the used `Criteria` that would additionally verify that this change does what it is expected to do. Does that make sense?

Reply:
Yes, it does. This tests seems to assume the resolution of the Smart Playlist rules works as expected, so it only checks if the nested playlist was updated or not. I think it is a valid test. 

IMO, what you are suggesting is more like an integration test, that checks all these parts working together, and we would need to setup data in the MediaFile repository for it to work. Of course it has its value, but requires a different setup. I'm personally ok with this test as is.

Reply:
More as a matter of learning about your testing principles in this project I would like to make the following comment. The test as written is looking at behavior internal to the unit under test, specifically the line including the conditional that the delay is included on. We're already in the data layer so I'm not sure scope is increased that much by touching another repo. At the same time, there are other ways to test this exact behavior such as ensuring that `refreshSmartPlaylist` is called on all relevant child playlists. In general I don't think this side-effect testing test is super valuable.

If this discussion has outlived its usefulness, please let me know. As I said I'm just trying to learn about how this project thinks about testing.